modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -149,11 +150,15 @@ def _construct_rsync_command(time: str, dest: str, recursive: bool = True, local
     raise Exception(f'Error constructing command. Either server was missing or {location} != {dest}')  # noqa: R503
 
 
-def _construct_git_pull(repo: str, submodules: bool = False) -> str:
+def _construct_git_pull(repo: str, submodules: bool = False, branch: Optional[str] = None) -> str:
     if submodules:
         extrap = '--recurse-submodules'
     else:
         extrap = ''
+
+    if branch:
+        extrap += f'--branch={branch}'
```

Comment:
Tests added. Please see output as --recurse-submodules & --branch don't work. That's why we do tests and this method is supposed to be easy to check.